processing/src/test/java/io/druid/segment/IndexMakerTest.java

```diff
@@ -0,0 +1,272 @@
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.io.Files;
+import io.druid.common.utils.JodaUtils;
+import io.druid.data.input.InputRow;
+import io.druid.data.input.MapBasedInputRow;
+import io.druid.granularity.QueryGranularity;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.CountAggregatorFactory;
+import io.druid.segment.data.CompressedObjectStrategy;
+import io.druid.segment.data.ConciseBitmapSerdeFactory;
+import io.druid.segment.incremental.IncrementalIndex;
+import io.druid.segment.incremental.OnheapIncrementalIndex;
+import org.apache.commons.io.FileUtils;
+import org.joda.time.DateTime;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(Parameterized.class)
+public class IndexMakerTest
+{
+  private static final long TIMESTAMP = DateTime.parse("2014-01-01").getMillis();
+  private static final AggregatorFactory[] DEFAULT_AGG_FACTORIES = new AggregatorFactory[]{
+      new CountAggregatorFactory(
+          "count"
+      )
+  };
+  private static final IndexSpec INDEX_SPEC = IndexMergerTest.makeIndexSpec(
+      new ConciseBitmapSerdeFactory(),
+      CompressedObjectStrategy.CompressionStrategy.LZ4
+  );
+  private static final List<String> DIMS = ImmutableList.of("dim0", "dim1");
+
+  private static final Function<Collection<Map<String, Object>>, Object[]> OBJECT_MAKER = new Function<Collection<Map<String, Object>>, Object[]>()
+  {
+    @Nullable
+    @Override
+    public Object[] apply(Collection<Map<String, Object>> input)
+    {
+      final ArrayList<InputRow> list = new ArrayList<>();
+      int i = 0;
+      for (final Map<String, Object> map : input) {
+        list.add(new MapBasedInputRow(TIMESTAMP + i++, DIMS, map));
+      }
+      return new Object[]{list};
+    }
+  };
+
+  @SafeVarargs
+  private static Iterable<Object[]> permute(Map<String, Object>... maps)
+  {
+    if (maps == null) {
+      return ImmutableList.<Object[]>of();
+    }
+    return Iterables.transform(
+        Collections2.permutations(
+            Arrays.asList(maps)
+        ),
+        OBJECT_MAKER
+    );
+  }
+
+  @Parameterized.Parameters
+  public static Iterable<Object[]> paramFeeder()
+  {
+    final Map<String, Object> map1 = new HashMap<>();
+    map1.put(DIMS.get(0), ImmutableList.<String>of("dim00", "dim01"));
+    map1.put(DIMS.get(1), "dim10");
```

Comment:
you can use ImmutableMap.of(k1, v1, k2, v2) to make those kinds of constructs more readable.